IA reader accessibility audit

develric opened the pull request:
Partially fixes #10962 (the items related to Reader directly)

**This PR is based on #11270 that must be merged before this one can be merged (currently targeting that PR branch to ease reviewing Files changed); UI design changes already approved there, so this Accessibility PR can be reviewed even if not merged before**

We made an audit of the Reader Subfilter component and Bottom Sheet using `Accessibility Scanner` and `TalkBack`:

| Subfilter Component | Reader Bottom Sheet |
|---|---|
| ![image](https://user-images.githubusercontent.com/47797566/74727111-efa25800-5240-11ea-8394-8ec885414089.png) | ![image](https://user-images.githubusercontent.com/47797566/74726977-b0740700-5240-11ea-8982-622518f4116a.png) |

## What we changed based on Accessibility Scanner Audit and TalkBack testing
1. Subfilter Component (`subfilter_component.xml`)
   a. Added contentDescription to relevant views and declaring `android:id="@+id/filter_list_button"` as not important for accessibility
   b. Increased the `android:id="@+id/remove_filter_button"` width to 48dp
2. Reader Bottom Sheet:
   a. Created focusable group in `subfilter_list_item.xml` and added relevant contentDescription so that the selected item in the bottom sheet is announced as selected from TalkBack
   b. Marked `android:id="@+id/empty_state_container"` as important for accessibility so to get the inner content announced directly

## NOTES
- We did not marked SITES and TAGS bottom sheet tabs as Headings since now they are tabs (in previous design we had a unique list of items)
- When no TAGS/SITES are present (like in the image below) we kept the text button height to the standard one I see we use across the app `@dimen/default_dialog_button_height` whereas we set a minWidth of `@dimen/min_touch_target_sz`
<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/47797566/74728534-3bee9780-5243-11ea-8f94-72b6764fe120.png">
</p>

## To test
Referring to the `What we changed based on Accessibility Scanner Audit and TalkBack testing` section above:
1. Subfilter Component: rerun Accessibility Scanner and check the above reported changes were effective
2. Reader Bottom Sheet: use talkback (tested with default navigation): use TalkBack to navigate the Bottom Sheet and check that the above reported changes were effective.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jd-alexander commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -1635,6 +1635,9 @@
     <string name="reader_btn_notifications_on">Turn on site notifications</string>
+    <string name="reader_btn_remove_filter">Remove the current filter</string>
+    <string name="reader_btn_select_filter">Select a Site or Tag to filter posts</string>
+    <string name="reader_btn_selected_filter_tick">Selected</string>
```
For these strings could you add `content_description` to the end of the name. Jayson recently created some localization guides to standardize this 😄 You can check it out [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/docs/localization.md)

develric replied:
Changed [here](5e9a55a57258c1c032713e4781b4b5b067a94afa) and I like the document 😊, thanks for pointing to it 👍.

jd-alexander replied:
Perfect!

jd-alexander commented on WordPress/src/main/res/values/dimens.xml:
```diff
@@ -170,7 +170,7 @@
 
     <dimen name="reader_filter_bottom_sheet_handle_height">15dp</dimen>
-    <dimen name="reader_remove_filter_icon_width">32dp</dimen>
+    <dimen name="reader_remove_filter_icon_width">48dp</dimen>
```
What do you think about removing this dimension altogether and utilizing `min_touch_target_sz` since it's the same?

develric replied:
Agree 👍! Changed [here](1e031da3cbdd5063fb448f7f8cdf0da86e412271)

jd-alexander commented on WordPress/src/main/res/layout/subfilter_list_item.xml:
```diff
@@ -3,7 +3,8 @@
     style="@style/SubfilterSiteTagItem"
-    android:layout_width="match_parent">
+    android:layout_width="match_parent"
+    android:focusable="true">
```
What was the reason for adding `focusable` here? Let me know. Thanks 😄

develric replied:
Oh you are right 😊! I initially was misusing it thinking I needed to have it there to announce the whole group and forgot to remove 🤦‍♂️. Thanks for catching, changed [here](7b2adcbbb78dddbfe3cf44092f4ef11c57631bad)

develric commented:
Hi @jd-alexander and thanks so much for helping with this 😊.

> I have a suggestion that could make the BottomSheet a bit clearer for the user...

Good point, sounds good to me. I changed it here 071354ac1075794fff06a46cf26947d54aca8b79 . I have put it in the `subfilter_bottom_sheet.xml` hope I got your intention right 😊

I also changed the target to be develop since #11270 was merged now.

I think I have followed up to all you comments. Let me know wdyt, thanks 🙇‍♂️

jd-alexander commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -1636,6 +1636,12 @@
+    <string name="reader_btn_remove_filter_content_description">Remove the current filter</string>
+    <string name="reader_btn_select_filter_content_description">Select a Site or Tag to filter posts</string>
+    <string name="reader_btn_selected_filter_tick_content_description">Selected</string>
+    <string name="reader_bottom_sheet_content_description">Select Tag or Site, Pop Up Window</string>
```
```suggestion
    <string name="reader_bottom_sheet_content_description">Select a Tag or Site, Pop Up Window</string>
```

develric replied:
👍 changed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/11333/commits/5d5e20b60334dab5060dc0157dd76d5131a74b17)
